native_client/ctcdecode/ctc_beam_search_decoder.cpp

```diff
@@ -149,6 +149,8 @@ decoder_next(const double *probs,
                        prefix_compare);
       for (size_t i = beam_size; i < state->prefixes.size(); ++i) {
         state->prefixes[i]->remove();
+        // Remove the element from std::vector
+        state->prefixes.erase(state->prefixes.begin() + i);
```

Comment:
This loop is erasing everything from beam_size to the end of the vector, so this can be a single erase call outside of the loop:

```C++
state->prefixes.erase(state->prefixes.begin() + beam_size, state->prefixes.end());
```

Reply:
Or alternatively:

```
state->prefixes.resize(beam_size);
```

Reply:
That's not completely wrong, I wrote it in a hurry ☺️